doc/httpinterceptors.md

```diff
@@ -0,0 +1,73 @@
+# HTTPInterceptors
+
+With the release of CDTDatastore 0.19 a new HTTP Interceptor API was introduced.
+HTTP Interceptors allow the developer to modify the HTTP requests and responses
+during the replication process.
+
+Interceptors can be used to implement your own authentication schemes, for
+example OAuth, or provide custom headers so you can perform your own analysis on
+usage. They can also be used to monitor the requests made by the library.
+
+To monitor or make changes to HTTP requests, conformance to the protocol `CDTHTTPInterceptor`
+is required and you need to implement one of the following methods:
+
+- To modify the outgoing request, `-interceptRequestInContext:`
+- To modify the incoming response, `-interceptResponseInContext:`
+
+An example of a HTTP interceptor:
+
+```objc
+#import "CDTHTTPInterceptor.h"
+
+@interface SampleInterceptor : NSObject <CDTHTTPInterceptor>
+
+@end
+
+@implementation SampleInterceptor
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        /* Init */
+    }
+    return self;
+}
+
+- (CDTHTTPInterceptorContext *)interceptRequestInContext:(CDTHTTPInterceptorContext *)context
+{
+    NSLog(@"Calling URL: %@", context.request.URL);
+
+    return context;
+}
+
+- (CDTHTTPInterceptorContext *)interceptResponseInContext:(CDTHTTPInterceptorContext *)context
+{
+    NSLog(@"Received response; URL: %@; status code: %@.",
+        context.request.URL, context.response.statusCode);
+    return context;
+}
+
+@end
+
+```
+
+In order to add an HTTP interceptor to a replication, you call the `-addInterceptor:` or
```

Comment:
I would prefer if this flowed on from the `SampleInterceptor` example rather than using `CookieInterceptor` which has not been introduced in the context of this document yet.